hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/server/events/EventQueue.java

```diff
@@ -79,17 +70,13 @@ public EventQueue(String threadNamePrefix) {
   // The field parent in DatanodeDetails class has the circular reference
   // which will result in Gson infinite recursive parsing. We need to exclude
   // this field when generating json string for DatanodeDetails object
-  static class DatanodeDetailsGsonExclusionStrategy
-          implements ExclusionStrategy {
-    @Override
-    public boolean shouldSkipField(FieldAttributes f) {
-      return f.getDeclaringClass() == NodeImpl.class
-              && f.getName().equals("parent");
-    }
-
-    @Override
-    public boolean shouldSkipClass(Class<?> aClass) {
-      return false;
+  public String serializeObject(Object obj) {
+    try {
+      return JsonUtils.toJsonString(obj);
+    } catch (Exception e) {
+      // Handle the exception, maybe log it
+      LOG.error("Error serializing object: ", e);
```

Comment:
To test the change in `EventQueue`, please run `ozone insight logs -v scm.event-queue` in an unsecure cluster (use `ozone` compose environment).

Errors related to serialization show up in SCM log:

```
scm_1       | 2024-04-09 06:08:32,299 [IPC Server handler 2 on default port 9861] ERROR events.EventQueue: Error serializing object: 
scm_1       | com.fasterxml.jackson.databind.exc.InvalidDefinitionException: No serializer found for class com.google.protobuf.UnknownFieldSet$Parser and no properties discovered to create BeanSerializer (to avoid exception, disable SerializationFeature.FAIL_ON_EMPTY_BEANS) (through reference chain: org.apache.hadoop.hdds.scm.server.SCMDatanodeHeartbeatDispatcher$NodeReportFromDatanode["report"]->org.apache.hadoop.hdds.protocol.proto.StorageContainerDatanodeProtocolProtos$NodeReportProto["unknownFields"]->com.google.protobuf.UnknownFieldSet["parserForType"])
scm_1       | 	at com.fasterxml.jackson.databind.exc.InvalidDefinitionException.from(InvalidDefinitionException.java:77)
scm_1       | 	at com.fasterxml.jackson.databind.SerializerProvider.reportBadDefinition(SerializerProvider.java:1330)
scm_1       | 	at com.fasterxml.jackson.databind.DatabindContext.reportBadDefinition(DatabindContext.java:414)
scm_1       | 	at com.fasterxml.jackson.databind.ser.impl.UnknownSerializer.failForEmpty(UnknownSerializer.java:53)
scm_1       | 	at com.fasterxml.jackson.databind.ser.impl.UnknownSerializer.serialize(UnknownSerializer.java:30)
scm_1       | 	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:732)
scm_1       | 	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:770)
scm_1       | 	at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:183)
scm_1       | 	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:732)
scm_1       | 	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:770)
scm_1       | 	at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:183)
scm_1       | 	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:732)
scm_1       | 	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:770)
scm_1       | 	at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:183)
scm_1       | 	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:502)
scm_1       | 	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:341)
scm_1       | 	at com.fasterxml.jackson.databind.ObjectMapper._writeValueAndClose(ObjectMapper.java:4793)
scm_1       | 	at com.fasterxml.jackson.databind.ObjectMapper.writeValueAsString(ObjectMapper.java:4038)
scm_1       | 	at org.apache.hadoop.hdds.server.JsonUtils.toJsonString(JsonUtils.java:70)
scm_1       | 	at org.apache.hadoop.hdds.server.events.EventQueue.serializeObject(EventQueue.java:75)
scm_1       | 	at org.apache.hadoop.hdds.server.events.EventQueue.fireEvent(EventQueue.java:195)
```

Reply:
These are the steps I followed to replicate:-

1. Started an unsecure ozone cluster on docker
2. Ran a freon command to create multiple files
3. While the files were getting created I executed the command ozone insight logs -v scm.event-queue

But I got No value present for it hence no parsing of any sort happened hence I could not see the error in SCM logs
Do you know how I can replicate it?

Reply:
This is coming when I execute it inside a docker container running SCM 
```
sh-4.2$ ozone insight logs -v scm.event-queue
No value present
sh-4.2$ 
```

Reply:
With the current patch the exception happens even without running `ozone insight`.  (See my other comment for the reason.)